os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -440,6 +440,43 @@ static void _up_assert(int errorcode)
 }
 
 #ifdef CONFIG_BINMGR_RECOVERY
+static void recovery_unblock_task(struct tcb_s *tcb)
+{
+	struct tcb_s *ntcb;
+
+	dq_rem((FAR dq_entry_t *)tcb, (dq_queue_t *)g_tasklisttable[tcb->task_state].list);
+
+	/* Add tcb to ready-to-run list */
+	if (sched_addprioritized(tcb, (FAR dq_queue_t *)&g_readytorun)) {
+		ASSERT(tcb->flink != NULL);
+		tcb->task_state = TSTATE_TASK_RUNNING;
+		tcb->flink->task_state = TSTATE_TASK_READYTORUN;
+	} else {
+		/* The tcb was added in the middle of the ready-to-run list */
+		tcb->task_state = TSTATE_TASK_READYTORUN;
+	}
+
+	/* Restore the exception context of the ntcb at the (new) head
+	 * of the g_readytorun task list.
+	 */
+	ntcb = this_task();
```

Comment:
Why do you name it as `ntcb`? How do you know it could be rtcb or ntcb?

Reply:
It means new tcb. This tcb is always new head of the g_readytorun list.
If fault thread is real-time, ntcb would be a real-time thread of other binary or fault message sender.
If not, it would be a fault message sender.